From 29a001ae82ec71a51fb6b03384a0a2c7689815f6 Mon Sep 17 00:00:00 2001
From: Du Chenyang <chenyang.du@intel.com>
Date: Mon, 16 Jul 2012 18:44:51 +0800
Subject: [PORT FROM MAIN] Adding build configuration for boot.bin

BZ: 46708

THIS SHOULD BE REPLACED BY https://android-secure.intel.com:8080/#/c/32/

Adding an option to support intel boot.bin build, which will disable
the default build of boot.img if TARGET_MAKE_NO_DEFAULT_BOOTIMAGE is true.

[BOOT CAMP SPECIFIC]

Category: device enablement
Domian: AOSP-Build
Origin: internal
Upstream-Candidate: yes

Change-Id: If4a85f757ca8eecce30542b9fd54e7652d1bed3f
Orig-Change-Id: I6b2cf1ad4f63517686085fa95ea6948153e2acbc
Signed-off-by: Sun Yijin <yijin.sun@intel.com>
Signed-off-by: Bruce Beare <bruce.j.beare@intel.com>
Signed-off-by: Du Chenyang <chenyang.du@intel.com>
Reviewed-on: http://android.intel.com:8080/74477
---
 core/Makefile |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/core/Makefile b/core/Makefile
index f118cea..0e0a0cc 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -416,6 +416,7 @@ ifdef BOARD_KERNEL_PAGESIZE
   INTERNAL_BOOTIMAGE_ARGS += --pagesize $(BOARD_KERNEL_PAGESIZE)
 endif
 
+ifneq ($(TARGET_MAKE_NO_DEFAULT_BOOTIMAGE),true)
 INSTALLED_BOOTIMAGE_TARGET := $(PRODUCT_OUT)/boot.img
 
 ifeq ($(TARGET_BOOTIMAGE_USE_EXT2),true)
@@ -446,6 +447,12 @@ bootimage-nodeps: $(MKBOOTIMG)
 
 endif # TARGET_BOOTIMAGE_USE_EXT2
 
+else # TARGET_MAKE_NO_DEFAULT_BOOTIMAGE
+
+$(INSTALLED_BOOTIMAGE_TARGET): $(MAKE_NO_DEFAULT_BOOTIMAGE_ITEMS)
+	$(MAKE_NO_DEFAULT_BOOTIMAGE)
+endif # TARGET_MAKE_NO_DEFAULT_BOOTIMAGE
+
 else	# TARGET_NO_KERNEL
 # HACK: The top-level targets depend on the bootimage.  Not all targets
 # can produce a bootimage, though, and emulator targets need the ramdisk
-- 
1.7.4.1

